jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/cm/processor/BaseFileProcessor.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2003 The Apereo Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *             http://opensource.org/licenses/ecl2
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sakaiproject.component.app.scheduler.jobs.cm.processor;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.sakaiproject.component.app.scheduler.jobs.cm.util.FileArchiveUtil;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.Map;
+
+@Slf4j
+public abstract class BaseFileProcessor extends BaseProcessor implements FileProcessor {
+
+    @Getter
+    @Setter
+    private String filename;
+
+    @Getter
+    @Setter
+    private int columns = -1;
+
+    @Getter
+    @Setter
+    private boolean headerRowPresent;
+
+    @Getter
+    @Setter
+    private boolean archive;
+
+    @Getter
+    @Setter
+    private FileArchiveUtil fileArchiveUtil;
+
+    public ProcessorState init(Map<String, Object> config) {
+        final String cName = this.getClass().getCanonicalName();
+
+        BaseFileProcessorState fps = new BaseFileProcessorState();
+        fps.setConfiguration(config);
+
+        String fName = (String) getPropertyOrNull(config, cName + ".filename");
+        String tempFileName = "";
+
+        if (StringUtils.isBlank(fName)) {
+            tempFileName = filename;
+            log.info("processor configured to load default file: {}", tempFileName);
+        } else {
+            String baseDir = (String) getPropertyOrNull(config, "path.base");
+            if (baseDir != null) {
+                tempFileName = baseDir;
+                if (!baseDir.endsWith(File.separator)) {
+                    tempFileName += File.separator;
+                }
+            }
+            tempFileName += fName;
+
+            log.info("processor configured to load override file: {}", tempFileName);
+        }
+
+        fps.setFilename(tempFileName);
+
+        int numCols = getColumns();
+
+        if (config.containsKey(cName + ".columns")) {
+            String cols = (String) config.get(cName + ".columns");
```

Comment:
Efficiency can be improved here. `containsKey()` and `get()` both have to search through the buckets. Instead, just do the `get()` outside the conditional and check the return value for not `null`.